### DISCUSSION TITLE
.print command

### DISCUSSION THREAD

**Discussion by JrgMyr** [pawelsalawa/sqlitestudio#4411] [Q&A]
sqlite3 has this non-standard feature to allow ".print" commands in multi-command-queries to show how far it got.  It seem like SQLiteStudio does not accept any dot commands.

Is there a way to output a line into the status pane just like ".print" would do?

**Comment by pawelsalawa** [comment]
You can create a custom SQL function, using QtScript as implemnation language, then in the body of the function type:
```
debug(arguments[0]);
```

Assuming you give this function name `print`, you should be able to use this statement in your SQL scripts:
```
SELECT print('my example debug string');
```
which would be equal to SQLite's `.print 'my example debug string'`.

**Reply by JrgMyr** [reply]
Could SQLiteStudio ignore any lines (or commands) that start with a dot such als ".header" or ".print"?
That way scripts could maintain no-error-compatibility between the orignial sqlite3 CLI and the Studio GUI.
If Studio regards a dot in the first text column as a comment that would already do the trick.

**Comment by JrgMyr** [comment]
Thant's an option. Thanks.
Is there any chance we could have the "native" `.print` command with script level compatibility between the CLI and the GUI?

**Reply by pawelsalawa** [reply]
It could be done, but it's a significant effort and the benefit is not that big, especially there are workaroundns, such as above. Hence I wouldn't treat it with high priority.

**Reply by JrgMyr** [reply]
Thanks for sharing your view on this.
